apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracerBuilder.java

```diff
@@ -118,6 +121,12 @@ public ElasticApmTracerBuilder withLifecycleListener(LifecycleListener listener)
         return this;
     }
 
+    // protected only for testing
+    public ElasticApmTracerBuilder withApmServerClient(ApmServerClient apmServerClient) {
```

Comment:
```suggestion
   ElasticApmTracerBuilder withApmServerClient(ApmServerClient apmServerClient) {
```

Reply:
It was actually required, thus this change was reverted in https://github.com/elastic/apm-agent-java/pull/2553/commits/7a718fbef47c6c8fd12be61e706e238e8078d6b4.

Reply:
So it's comment is wrong 🙂

Reply:
But the correct thing to do is leave it package-private.
The way I dealt with providing test access like that is through `TracerInternalApiUtils`. I'd encourage you to do that (even use the same class as it is in the same package) instead of making a test-only method `public`